core/cmd/app.go

```diff
@@ -187,16 +197,44 @@ func NewApp(client *Client) *cli.App {
 				},
 			},
 			Before: func(c *cli.Context) error {
-				if client.flagsProcessed {
-					if c.IsSet("config") || c.IsSet("secrets") {
-						// invalid mix of flags here and root
-						return fmt.Errorf("multiple commands with --config or --secrets flags. only one command may specify these flags. when secrets are used, they must be specific together in the same command")
+				errNoDuplicateFlags := fmt.Errorf("multiple commands with --config or --secrets flags. only one command may specify these flags. when secrets are used, they must be specific together in the same command")
+				if c.IsSet("config") {
+					if client.configFilesIsSet || client.secretsFileIsSet {
+						return errNoDuplicateFlags
+					} else {
+						client.configFiles = c.StringSlice("config")
 					}
-					// flags at root
-					return nil
 				}
+
+				if c.IsSet("secrets") {
+					if client.configFilesIsSet || client.secretsFileIsSet {
+						return errNoDuplicateFlags
+					} else {
+						client.secretsFile = c.String("secrets")
+					}
+				}
+
 				// flags here, or ENV VAR only
-				return client.initConfigAndLogger(&opts, c.StringSlice("config"), c.String("secrets"))
+				err := client.initServerConfig(&opts, client.configFiles, client.secretsFile)
+				if err != nil {
+					return err
+				}
+
+				if (client.Config.LogFileMaxSize() / utils.MB) > 0 {
+					err = utils.EnsureDirAndMaxPerms(client.Config.LogFileDir(), os.FileMode(0700))
+					if err != nil {
+						return err
+					}
+				}
+
+				// Swap out the logger, replacing the old one.
+				client.CloseLogger()
+
+				logger := client.Config.NewLogger()
+				client.Logger = logger
```

Comment:
aside: are we using the word `client` here to mean `server`? the (excellent, thank you) comment earlier in the file says something like  ~ 'server can swap out the logger for a rotating logger'

Reply:
IIUC the `client` type supports both - definitely could use a better name